File: go-version/leetcode/LC1925/count_triples.go
package LC1925

import "math"

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/12/8 上午8:23
 * @FilePath: count_triples
 * @Description: 枚举遍历
 */

func countTriples(n int) int {
	var ans int
	for a := 1; a < n; a++ {
		for b := 1; b < a && a*a+b*b <= n*n; b++ {
			c := a*a + b*b
			rt := int(math.Sqrt(float64(c)))
			if rt*rt == c {
				ans++
			}
		}
	}
	return ans * 2
}

File: python-version/LC1925/count_triples.py
import math


class Solution:
    @staticmethod
    def countTriples(n: int) -> int:
        ans = 0
        for a in range(1, n):
            for b in range(1, a):
                c = a * a + b * b
                if a * a + b * b > n * n:
                    break
                rt = int(math.sqrt(c))
                if rt ** 2 == c:
                    ans += 1
        return ans * 2
